### PR TITLE
chore: integrate ongoing-rewards-cards

### PR DESCRIPTION
**Pull request by dennyscode**
Ticket: https://lifi.atlassian.net/browse/LF-10177

### CODE REVIEW

**Review by tcheee — APPROVED** [review]

File: src/components/ProfilePage/QuestCarousel/QuestCarousel.tsx [hunk 1]
```diff
@@ -1,89 +1,105 @@
...
-  loading,
-  pastCampaigns,
-}: QuestCarouselProps) => {
-  const { url } = useOngoingQuests();
+export const QuestCarousel = ({ pastCampaigns }: QuestCarouselProps) => {
   const { t } = useTranslation();
-
+  const { quests, isLoading: isQuestsLoading, url } = useOngoingQuests();
+  const { data: ongoingRewards, isLoading: isOngoingRewardsLoading } =
+    useOngoingRewards();
```
Comment (tcheee) [inline src/components/ProfilePage/QuestCarousel/QuestCarousel.tsx:20]: rename to `useOngoingNumericQuests` pls and `ongoingNumericQuests`

File: src/components/ProfilePage/QuestCarousel/QuestCarousel.tsx [hunk 2]
```diff
@@ -1,89 +1,105 @@
...
+    !isNotLive && (
+      <QuestCarouselContainer>
+        <CarouselContainer title={t('missions.available')}>
+          {loading ? (
+            Array.from({ length: 3 }, (_, idx) => (
+              <QuestCardSkeleton key={`mission-card-skeleton-${idx}`} />
+            ))
+          ) : (
+            <>
+              {quests?.map((quest: Quest, index: number) =>
```
Comment (tcheee) [inline src/components/ProfilePage/QuestCarousel/QuestCarousel.tsx:91]: not sure about this logic, feels weird. @oktapodia any viewpoint?

File: src/hooks/useOngoingRewards.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    if (account.address === storedAddress) {
+      return;
+    }
+
+    reset();
+  }, [account, reset, storedAddress]);
+
+  //we store the data during 24hours to avoid querying too much our partner API.
```
Comment (tcheee) [inline src/hooks/useOngoingRewards.ts:49]: ?

File: src/hooks/useOngoingRewards.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  displayName: string;
+  level?: number;
+  id: number;
+  image: string;
+  name: string;
+  nextLevel: number;
+  type: string;
+}
+
+export const useOngoingRewards = (): UseOngoingRewardsType => {
```
Comment (tcheee) [inline src/hooks/useOngoingRewards.ts:29]: to rename

File: src/components/ProfilePage/QuestCarousel/QuestCarousel.tsx (hunk 1 above)
Comment (dennyscode) [reply src/components/ProfilePage/QuestCarousel/QuestCarousel.tsx]: I did rename it but I applied that name to stay consistent with backend where the endpoint is called `ongoing-rewards`.

File: src/components/ProfilePage/QuestCarousel/QuestCarousel.tsx (hunk 2 above)
Comment (dennyscode) [reply src/components/ProfilePage/QuestCarousel/QuestCarousel.tsx]: My render-functions were an attempt to split logic from render function. I am still not sure what´s best, but I removed those functions for now.

File: src/hooks/useOngoingRewards.ts (hunk 4 above)
Comment (dennyscode) [reply src/hooks/useOngoingRewards.ts]: Done

File: src/components/ProfilePage/QuestCardDetailled/QuestCardDetailled.tsx [hunk 5]
```diff
@@ -174,7 +189,7 @@ export const QuestCardDetailled = ({
                     lineHeight="18px"
                     color={'#ffffff'}
                   >
-                    {`${points}`}
+                    {`${formatDecimal(points)}`}
```
Comment (tcheee) [inline src/components/ProfilePage/QuestCardDetailled/QuestCardDetailled.tsx:192]: we don't have decimals on points, is it necessary?

File: src/components/ProfilePage/QuestCarousel/QuestCarousel.tsx [hunk 6]
```diff
@@ -1,89 +1,104 @@
...
-                        quest?.attributes.Points > 0 && rewardType === 'weekly'
-                      }
-                      rewardRange={rewardRange}
-                    />
-                  );
-                })
-              : Array.from({ length: 3 }, () => 42).map((_, idx) => (
-                  <QuestCardSkeleton key={'mission-card-skeleton-' + idx} />
+              {/** render ongoing numeric quests */}
+              {ongoingNumericQuests &&
```
Comment (tcheee) [inline src/components/ProfilePage/QuestCarousel/QuestCarousel.tsx:82]: what do you think about extracting this in its own component @dennyscode @oktapodia ?

File: src/components/ProfilePage/QuestCardDetailled/QuestCardDetailled.tsx (hunk 5 above)
Comment (dennyscode) [reply src/components/ProfilePage/QuestCardDetailled/QuestCardDetailled.tsx:192]: We had points with many decimals which looked odd: This "formatDecimals" reduces decimals to 2. Wdyt?

File: src/components/ProfilePage/LevelBox/ProgressionBar.tsx [hunk 7]
```diff
@@ -23,7 +28,21 @@ export const ProgressionBar = ({ points, levelData }: ProgressionBarProps) => {
       : 0;
 
   return (
-    <Box>
+    <Box
```
Comment (oktapodia) [inline src/components/ProfilePage/LevelBox/ProgressionBar.tsx:31]: Would prefer a styled component

File: src/utils/formatDecimals.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+export function formatDecimal(number: number) {
+  if (!number) {
+    return;
+  }
+  return Number.isInteger(number)
```
Comment (oktapodia) [inline src/utils/formatDecimals.ts:5]: what is the goal of that helper?

File: src/components/OptionalLink.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  alt?: string;
+  styles?: CSSObject;
+}
+
+export const OptionalLink: React.FC<PropsWithChildren<OptionalLinkProps>> = (
+  props,
+) => {
+  if (props.url) {
+    return (
+      <NextLink
```
Comment (oktapodia) [inline src/components/OptionalLink.tsx:16]: can you do it as mui component so we keep theming? e.g. https://github.com/jumperexchange/jumper-exchange/pull/1344/files#diff-927fb0b729f8d1aaff94ec40b4bfe2d47f24e69dd48e9fb1fb2ba502028ba91eR145

Also, url -> href (keep same name for consistency)
styles -> sx

etc..

**Review by oktapodia — CHANGES_REQUESTED** [review]
Couple of comments

File: src/components/OptionalLink.tsx (hunk 9 above)
Comment (dennyscode) [reply src/components/OptionalLink.tsx:16]: Good idea! Done

File: src/utils/formatDecimals.ts (hunk 8 above)
Comment (dennyscode) [reply src/utils/formatDecimals.ts]: It was meant to be a fallback if we run into numbers with multiple decimals.

Comment (dennyscode) [reply src/utils/formatDecimals.ts]: I have removed it again

File: src/components/ProfilePage/LevelBox/ProgressionBar.tsx (hunk 7 above)
Comment (dennyscode) [reply src/components/ProfilePage/LevelBox/ProgressionBar.tsx]: Indeed! Good idea, done

**Review by oktapodia — APPROVED** [review]
LGTM

**Pull request merged**